From bafcea78f5615d24185cc07832ee3ef2ce01f0e2 Mon Sep 17 00:00:00 2001
From: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
Date: Mon, 1 Aug 2022 10:27:42 +0530
Subject: [PATCH] mac80211: Fix the ht_capa check warning

In sta_ht_capa_read() we allocate the memory for ht_capabilies
buffer size for print. Based on this size we move the buffer
pointer and pass the remaining size to scnprintf for printing. 
In problematic case, we calculating the remaining size by sizeof (buf) 
instead of allocated buffer size in PRINT_HT_CAP define. 

p += scnprintf(p, sizeof (buf) + buf - p, "\t" _str "\n");
(i.e) sizeof (buf) + buf - p = 8 + 1000 - 1200 = -ive value passing to scnprintf. 
So only observing Warning.

Now i changed the remaining size calculation based on allocated buffer size.

Signed-off-by: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
---
 net/mac80211/debugfs_sta.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/debugfs_sta.c b/net/mac80211/debugfs_sta.c
index 4e2a364..39adfac 100644
--- a/net/mac80211/debugfs_sta.c
+++ b/net/mac80211/debugfs_sta.c
@@ -441,7 +441,7 @@ static ssize_t sta_ht_capa_read(struct file *file, char __user *userbuf,
 #define PRINT_HT_CAP(_cond, _str) \
 	do { \
 	if (_cond) \
-			p += scnprintf(p, sizeof(buf)+buf-p, "\t" _str "\n"); \
+			p += scnprintf(p, bufsz + buf - p, "\t" _str "\n"); \
 	} while (0)
 	char *buf, *p;
 	int i;
-- 
2.17.1

